docs/visualizing-your-sql-queries/plotting-with-ggplot.md

```diff
@@ -1,14 +1,638 @@
+
+## Load Engine
+
+We now load in our SQL extension that allows us to execute SQL queries in Jupyter Notebooks.
+
+```{important}
+<b>Note</b> Ensure you restart any previous notebook that has the same database name as the one initialized below.
+```
+
+```{code-cell} ipython3
+# Loading in SQL extension
+%load_ext sql
+# Initiating a DuckDB database named 'bank_data.duck.db' to run SQL queries
+%sql duckdb:///bank_data.duck.db
+```
+
+<!-- endregion -->
+
+## Creating Tables
+
+Let's start off with loading three of the eight `.csv` files from the `expanded_data` folder in the current directory to our newly created DuckDB database. Like in the previous tutorial, we will [create a schema](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#creating-a-schema) `s1` in which we will store the tables. Here we use the `CREATE TABLE` syntax in DuckDB to ingest four of the eight `.csv` files. The `read_csv_auto` is a function that helps SQL understand our local `.csv` file for creation into our database.
+
+```{code-cell} ipython3
+%%sql
+CREATE SCHEMA s1;
+CREATE TABLE s1.account AS
+FROM read_csv_auto('expanded_data/account.csv', header=True, sep=',');
+CREATE TABLE s1.district AS
+FROM read_csv_auto('expanded_data/district.csv', header=True, sep=',');
+CREATE TABLE s1.loan AS
+FROM read_csv_auto('expanded_data/loan.csv', header=True, sep=',');
+```
+
+The code above will create three tables in the database schema: `s1.account`, `s1.district`, `s1.loan`.
+
+## Exploring the data
+
+Let's take a look at its entries.
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.account
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.district
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.loan
+```
+
+## `%sqlplot`
+
+The `%sqlplot` magic command is the easiest JupySQL plotting command to learn. Currently, it supports four different types of visualizations, including `boxplot`, `histogram`, `bar`plot, and `pie`chart. All `%sqlplot` visualizations return customizable [`matplotlib.Axes` objects](https://ploomber-sql.readthedocs.io/en/latest/visualizing-your-sql-queries/plotting-with-seaborn.html#axes-level) and have three magic commands in common:
+
+- `--table/-t` Table to use
+
+- `--column/-c` Column(s) to plot. <b>Note</b> only `boxplot` and `histogram` can accept multiple columns
+
+- `--with/-w` Use a previously saved query (CTE) as input data
+
+There are other magic commands unique to the aforementioned visualizations as well. Let us learn from a few examples below.
+
+### Examples
+
+1. Suppose the finance manager wants to visualize boxplots of `average_salary` and loan `payments` in a single graph. We will, first, join all three tables to obtain the relevant data, save the output as a CTE, and use that CTE for `%sqlplot boxplot`:
```

Comment:
"We will, first, join all three tables to obtain the relevant data," -> "We will first join all three tables to obtain the relevant data,"

Reply:
resolved, good to close this comment